vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridAllowNestedNullsIT.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2000-2018 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.grid.it;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.vaadin.flow.component.grid.testbench.GridElement;
+import com.vaadin.flow.testutil.AbstractComponentIT;
+import com.vaadin.flow.testutil.TestPath;
+
+@TestPath("allow-nested-nulls")
+public class GridAllowNestedNullsIT extends AbstractComponentIT {
```

Comment:
Just to confirm that the current behavior is not changed, there could be a test that expects that a `NPE` is thrown when `NestedNullBehavior.THROW` is used. It could used the the same page with a button that changes the behavior and resets the `DataProvider` when clicked.